objects/src/transaction/tx_script.rs

```diff
@@ -1,11 +1,55 @@
 use super::{Digest, Felt, Word};
 use crate::{
     assembly::{Assembler, AssemblyContext, ProgramAst},
+    notes::NoteId,
     utils::collections::{BTreeMap, Vec},
     vm::CodeBlock,
     TransactionScriptError,
 };
 
+// TRANSACTION ARGS
+// ================================================================================================
+
+/// A struct that represents optional transaction arguments.
+///
+/// The [TransactionArgs] object is composed of:
+/// - [struct](TransactionScript): a program that is executed in a transaction after all input
+///   notes have been executed..
+/// - [BTreeMap](NoteArgs): data being put on stack when the note script is executed. Different
+///   to Note Inputs, Note Args can be used by the executing account.
+#[derive(Clone, Debug, Default)]
+pub struct TransactionArgs {
+    tx_script: Option<TransactionScript>,
+    note_args: Option<BTreeMap<NoteId, Word>>,
+}
```

Comment:
I think this was supposed be be:
```Rust
pub struct TransactionArgs {
    tx_script: Option<TransactionScript>,
    note_args: BTreeMap<NoteId, Word>,
}
```

Reply:
Not sure if I follow. `NoteArgs` are optional to a transaction the same way as a `TransactionScript`, right? So the canonical way to indicate this optionality would be 

```Rust
pub struct TransactionArgs {
    tx_script: Option<TransactionScript>,
    note_args: Option<BTreeMap<NoteId, Word>>,
}
``` 

Why would you put `NoteArgs` not as an option? `note_args: BTreeMap<NoteId, Word>`

Reply:
An empty `BTreeMap` is functionally equivalent to `None` in this case but without the extra step we need to take to unwrap it.